### PR TITLE
remove warnings about unused arguments

### PR DESCRIPTION
**Pull request by kalj**
Just removing some unnecessary warnings.

### CODE REVIEW

File: source/base/function_spherical.cc [hunk 1]
```diff
@@ -225,7 +225,7 @@ namespace Functions
   template <int dim>
   SymmetricTensor<2,dim>
   Spherical<dim>::hessian (const Point<dim> &p_,
-                           const unsigned int component) const
+                           const unsigned int /* component */) const
```
Comment (bangerth) [inline source/base/function_spherical.cc]: Oh, this is a bug -- the component should actually be passed to `sgradient()` and `shessian()` below (line 232/233). Do you agree, @davydden ?

Comment (davydden) [inline source/base/function_spherical.cc]: @bangerth no, i don't ;-) this is a scalar-valued function, so `component` is unused.

Comment (bangerth) [inline source/base/function_spherical.cc]: But you added the `component` argument to the `svalue()`, `sgradient()`, and `shessian()` functions. Why not just pass the component through to these functions? Right now, these arguments are all unused. (I see now that the `value()` and `gradient()` functions also ignore the component...)

Comment (davydden) [inline source/base/function_spherical.cc]: no, i did not add those argument to `svalue()/sgradient()/shessian()`. One probably could pass it through, but this was not my intention and I have not though this case through.

**Comment by davydden** [comment]
related to https://github.com/dealii/dealii/pull/3875

File: source/base/function_spherical.cc (hunk 1 above)
Comment (bangerth) [inline source/base/function_spherical.cc]: Hm, you are correct -- these arguments indeed do not exist... Must be the altitude here that made me think that they do...

**Pull request merged**